### PR TITLE
[Hapi] Updated timeout property of RoutePayloadConfigurationObject

### PR DESCRIPTION
**Pull request by chamini2**
RoutePayloadConfigurationObject is an optional property which accepts a `number` or `false`.

As stated in the [docs](https://hapijs.com/api#route-options)

> timeout - payload reception timeout in milliseconds. Sets the maximum time allowed for the client to transmit the request payload (body) before giving up and responding with a Request Timeout (408) error response. Set to false to disable. Defaults to 10000 (10 seconds).

### CODE REVIEW

File: types/hapi/index.d.ts [hunk 1]
```diff
@@ -464,7 +464,9 @@ export class Server extends Podium {
      * If no callback is provided, a Promise object is returned.
      * [See docs](https://hapijs.com/api/16.1.1#serverstopoptions-callback)
      * @param options  options object with:
-     *      * timeout - overrides the timeout in millisecond before forcefully terminating a connection. Defaults to 5000 (5 seconds).
+     *      * 
+     
+- overrides the timeout in millisecond before forcefully terminating a connection. Defaults to 5000 (5 seconds).
```
Comment (AJamesPhillips) [inline types/hapi/index.d.ts]: Could you revert to the previous formatting please.

File: types/hapi/index.d.ts [hunk 2]
```diff
@@ -1025,7 +1027,7 @@ export interface RoutePayloadConfigurationObject {
     /** limits the size of incoming payloads to the specified byte count. Allowing very large payloads may cause the server to run out of memory. Defaults to 1048576 (1MB). */
     maxBytes?: number;
     /** payload reception timeout in milliseconds. Sets the maximum time allowed for the client to transmit the request payload (body) before giving up and responding with a Request Timeout (408) error response. Set to false to disable. Defaults to 10000 (10 seconds). */
-    timeout?: number;
+    timeout?: number | false;
```
Comment (AJamesPhillips) [inline types/hapi/index.d.ts]: Excellent, thanks.  Can you add a test as well please?

**Review by AJamesPhillips — CHANGES_REQUESTED** [review]
Have requested some minor changes, thanks for the contribution.

File: types/hapi/index.d.ts (hunk 1 above)
Comment (chamini2) [inline types/hapi/index.d.ts]: Yes, I'm on it!

File: types/hapi/index.d.ts (hunk 2 above)
Comment (chamini2) [inline types/hapi/index.d.ts]: Done.

**Comment by chamini2** [comment]
Does anyone know when do we get a release of this?

**Comment by AJamesPhillips** [comment]
@chamini2 once merged it's merged with master it should automatically be published to: https://www.npmjs.com/package/@types/hapi which it looks like it has as the updated date is: Mon, 21 Aug 2017 21:55:03 GMT

**Pull request merged**